crypto/x509/v3_asid.c

```diff
@@ -689,15 +689,30 @@ static int asid_contains(ASIdOrRanges *parent, ASIdOrRanges *child)
  */
 int X509v3_asid_subset(ASIdentifiers *a, ASIdentifiers *b)
 {
-    return (a == NULL ||
-            a == b ||
-            (b != NULL &&
-             !X509v3_asid_inherits(a) &&
-             !X509v3_asid_inherits(b) &&
-             asid_contains(b->asnum->u.asIdsOrRanges,
-                           a->asnum->u.asIdsOrRanges) &&
-             asid_contains(b->rdi->u.asIdsOrRanges,
-                           a->rdi->u.asIdsOrRanges)));
+    int subset;
+
+    if (a == NULL || a == b)
+        return 1;
+
+    if (b == NULL)
+        return 0;
+
+    if (X509v3_asid_inherits(a) || X509v3_asid_inherits(b))
+        return 0;
+
+    subset = (a->asnum == NULL && b->asnum == NULL)
```

Comment:
I think requiring  `&& b->asnum == NULL` here is too stringent. If `a` only specifies RDI resources, but no AS numbers, it is still a subset of `b` even if `b` happens to specify resources of both type.

The analogous check in `X509v3_addr_subset()` considers an `a` only specifying IPv4 addresses a subset of a `b` specifying IPv4 and IPv6 addresses.

Reply:
Good point. Fixed.